src/actions/insightActions.ts

```diff
@@ -60,4 +83,22 @@ export const storeInsightMedia = (imageFile: File): string | null => {
             console.log("Error - FirebaseStorage - Upload Insight files", error);
         });
     return null;
+}
+
+export const getInsights = async (organization: string, group: string): Promise<{[id: string]: Insight}> => {
+    const groupRef: CollectionReference =
+        collection(firestore, `${ORGANIZATIONS_COLLECTION}/${organization}/${GROUPS_COLLECTION}/${group}/${INSIGHTS_COLLECTION}`);
+    const queryRequest: Query = query(groupRef, orderBy("lastUpdated", "desc"), limit(QUERY_LIMIT));
+
+    const insights: {[id: string]: Insight} = {};
+
+    try {
+        (await getDocs(queryRequest)).forEach((insightDocument) => {
+            insights[insightDocument.id] = insightDocument.data() as Insight;
```

Comment:
I'm generally against typecasting like this, we could create an InsightDocument type with
`data: Function<Insight>` and assign it inside the forEach but since this is firebase I guess it's more of a style preference

Reply:
Made a cursory attempt at this but ran into some issues. I feel this could be pushed to a dedicated CR to add typing to all of the firebase requests